File: CodingTest/Java/Solution4.java
// https://school.programmers.co.kr/learn/courses/30/lessons/70129
package CodingTest.Java;

public class Solution4 {
    public int[] solution(String s) {
        int[] answer = {};
        int remove0 = 0;
        int times = 0;
        while (true) {
            if (s.equals("1")) {
                break;
            }
            times += 1;
            int ogLen = s.length();
            s = s.replace("0","");
            int tempS = s.length();
            remove0 += ogLen - tempS;
            String binaryString = Integer.toBinaryString(tempS);
            s = binaryString;
        }
        answer = new int[] {times, remove0};
        return answer;
    }
}

File: CodingTest/Python/solution4.py
# https://school.programmers.co.kr/learn/courses/30/lessons/42578
def solution(clothes):
    answer = 1
    lst = {}
    for cloth in clothes:
        if cloth[1] not in lst.keys():
            lst[cloth[1]] = 1
        else:
            lst[cloth[1]] += 1

    lst = list(lst.values())
    for l in lst:
        answer *= (l+1)
    return answer - 1